/imu_ekf_node:
  
  imu_topic: "/imu/data_newIMU"

  noise_filter: true
  fc: 25.0  # cut-off frequency of noise
  fs: 100.0  # sample frequency of IMU

  use_magnetometer: true
  mag_topic: "/mag_newIMU"

  mag_bias_x: 0.0  # Hard iron bias x (µT)
  mag_bias_y: 0.0  # Hard iron bias y (µT)
  mag_bias_z: 0.0  # Hard iron bias z (µT)
  acc_noise: 0.25  # Accelerometer measurement noise variance (m²/s⁴, σ_a² = 0.5²)
  mag_noise: 0.01  # Magnetometer measurement noise variance (µT², σ_m² = 0.1²)
  gyro_noise: 0.09  # Gyroscope process noise variance (rad²/s², σ_ω² = 0.3²)
  magnetic_reference_x: 29.14  # Magnetic field reference x (µT, Gongju-si, NED)
  magnetic_reference_y: -4.46  # Magnetic field reference y (µT, Gongju-si, NED)
  magnetic_reference_z: 45.00  # Magnetic field reference z (µT, Gongju-si, NED)
  coordinate_frame: "NED"  # Coordinate frame: "NED" or "ENU"